### PR TITLE
Support culture in content Saving and Publishing events

### PR DESCRIPTION
**Pull request by zpqrtbnk**
Connects #3640

### CODE REVIEW

File: src/Umbraco.Tests/Services/ContentServiceEventTests.cs [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+            {
+                var saved = e.SavedEntities.First();
+
+                Assert.AreSame(document, saved);
+
+                Assert.IsTrue(saved.HasSavedCulture("fr-FR"));
+                Assert.IsFalse(saved.HasSavedCulture("en-UK"));
+            }
+
+            ContentService.Saving += OnSaving;
```
Comment (Shazwazza) [inline src/Umbraco.Tests/Services/ContentServiceEventTests.cs]: Maybe should be in a try/finally?

File: src/Umbraco.Core/Models/Content.cs [hunk 2]
```diff
@@ -285,6 +282,24 @@ internal void SetPublishInfo(string culture, string name, DateTime date)
             _publishInfos.AddOrUpdate(culture, name, date);
         }
 
+        // internal for repository
+        internal void AknPublishInfo()
```
Comment (Shazwazza) [inline src/Umbraco.Core/Models/Content.cs]: What does `Akn` mean?

Comment (zpqrtbnk) [reply src/Umbraco.Core/Models/Content.cs]: aknowledge

**Comment by zpqrtbnk** [comment]
I am entirely happy with it all and it passes tests - I have done some cleanup too - now there is one `//fixme` left in `Content` and I don't know what to do with it. If you can get rid of it, then please merge, else tell me what should be done with it?

**Pull request merged**